Move characterization to the FPR

mistydemeo opened the pull request:
This moves characterization to the FPR, allowing characterization rules to be defined and customized. Two new categories of FPR rules are added "characterization" and "default characterization".

FITS no longer runs on every file, and the old archivematicaFITS.py no longer runs at all. "Default characterization" rules will be run on any tool that doesn't have a specific characterization tool, and FITS will be the default shipped by Archivematica. An arbitrary number of characterization rules can be defined for a given format.

FPR database dump updates aren't covered here - FPR rules will get taken care of post-merge.

fiwalk will be a new characterization rule included when this is merged, and some fiwalk support files are included.

This depends on artefactual/archivematica-fpr-admin#1 in order for the new rules to show up in the UI.

Commit:
Move characterization to FPR rules

Commit:
Characterization: add default commands

One or more default normalization commands can be defined, by defining
FPRules with the purpose 'default_characterization'. As with
normalization, these will be called if no matching FPRules can be found.
This also allows for characterization rules to be applied to files that
couldn't be identified.

FITS will be the default characterization command in 1.2; others can
be introduced, or customized by users.

mistydemeo committed:
Characterization: add fiwalk plugins

Hwesta commented on src/MCPClient/lib/clientScripts/characterizeFile.py:
```diff
@@ -0,0 +1,76 @@
+    if not rules:
+        rules = FPRule.active.filter(purpose='default_characterization')
+
+    for rule in rules:
```
Confirming - we want to run _all_ the characterization rules we find?

mistydemeo replied:
Yes: the idea was that multiple characterization tools could be defined for a given format, and all should be run. Same is also true for default characterization.

Hwesta commented:
We'll have to address 4f499d5b1a5c2da3a51e9a0b1c3fcc1a84da397a / https://projects.artefactual.com/issues/6160 / https://projects.artefactual.com/issues/6161 before we can drop archivematicaFITS entirely :(  I think that's blocking on this commit until we figure out what goes into the METS there.

Also, I don't know anything about the fiwalk plugins, so I can't comment on them.

Other than that, looks good!

Hwesta commented on src/MCPClient/lib/clientScripts/characterizeFile.py:
```diff
@@ -0,0 +1,76 @@
+if __name__ == '__main__':
+    file_path = sys.argv[1]
+    file_uuid = sys.argv[2]
+    exit(main(file_path, file_uuid))
```
sys.exit?

mistydemeo replied:
That keeps tripping me up!

mistydemeo commented:
> We'll have to address 4f499d5 / https://projects.artefactual.com/issues/6160 / https://projects.artefactual.com/issues/6161 before we can drop archivematicaFITS entirely :(

That needs to get moved in to here either way - archivematicaFITS isn't going to be run under this! I'll talk to Evelyn.

Commit:
createMETS2: correct FPCommandOutput purposes

Commit:
createMETS2: fix FPCommandOutput with encoding declarations

mistydemeo commented:
Caught a couple minor bugs and pushed fixes. Will squash when merging.

The lxml/unicode one was a nasty surprise.

Commit:
FPCommandOutput: use combo of file and rule UUIDs for pk

Previously only one output per FPCommand was allowed, whereas we want
to accept one per file/command combination.

Commit:
transcoder: Add a few stdout separators

Commit:
RD: add new to_gnu_options() method

This makes it easier for commands that pass options to non-shell script
commands to do so as CLI arguments.

Hwesta commented:
Looks good.  Can't comment on the fiwalk plugin changes, and the parts that are in other pull requests have comments there.

mistydemeo commented:
Rebased and squashed; I think this is ready.

Commit:
Convert submission documentation workflow to use new ID tool

Hwesta commented:
Still looks good

Commit:
Remove duplicate chain post-normalization

Post-normalization, a full chain is repeated twice. The only difference
between the two chains is a variable set at the very end; otherwise
every chainlink is identical.

The duplicated chains were making it difficult to introduce new
microservices that occured in this part of the workflow, especially
when introducing new choices.

This commit unifies the two chains. They branch at the very beginning,
to set the unit variable, and then converge into an identical set of
chainlinks.

mistydemeo commented:
The commit history here is messy because it's been rebased on top of #22, but otherwise I think this is good to go.

mistydemeo commented:
Rebased on top of qa/1.x now that #22 is merged. I'd like to get this merged ASAP to make sure that client branches that need this are okay.
